File: src/app/gkill/api/req_res/update_kmemo_request.go
package req_res

import "github.com/mt3hr/gkill/src/app/gkill/dao/reps"

type UpdateKmemoRequest struct {
	SessionID string `json:"session_id"`

	Kmemo reps.Kmemo `json:"kmemo"`

	TXID *string `json:"tx_id"`

	LocaleName string `json:"locale_name"`

	UpdatedKyou *reps.Kyou `json:"updated_kyou"`

	WantResponseKyou bool `json:"want_response_kyou"`
}

File: src/client/classes/api/req_res/update-kmemo-request.ts
'use strict'

import { Kmemo } from '@/classes/datas/kmemo'
import { GkillAPIRequest } from '../gkill-api-request'
import { Kyou } from '@/classes/datas/kyou'

export class UpdateKmemoRequest extends GkillAPIRequest {

    kmemo: Kmemo

    tx_id: string | null = null

    want_response_kyou: boolean

    updated_kyou: Kyou | null | null = null

    constructor() {
        super()
        this.kmemo = new Kmemo()
        this.want_response_kyou = false
    }

}


